### PR TITLE
Move `forc`'s block explorer support into a dedicated `forc-explore` plugin crate

### LINKED ISSUE #946: Move `forc explore` command into a dedicated `forc-explore` executable crate
We should implement `forc explore` as a forc plugin as described in #944.

### PR DESCRIPTION
**Pull request by mitchmindtree**
**NOTE: This PR is Based on https://github.com/FuelLabs/sway/pull/1178.**

I've taken the liberty to clean-up/refactor the code a little during the move too.

Closes #946.

### CODE REVIEW

**Review by JoshuaBatty — DISMISSED** [review]
nice, looks good.

File: forc-explore/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+[package]
+name = "forc-explore"
+version = "0.1.0"
```
Comment (adlerjohn) [inline forc-explore/Cargo.toml:3]:
```suggestion
version = "0.9.2"
```

**Review by adlerjohn — COMMENTED** [review]
While we're here, could you use major.minor for non-Fuel deps in this new crate instead of major.minor.patch?

**Review by sezna — APPROVED** [review]
once again, mostly just moved code right? nice, loving the plugin system. this is awesome

**Review by JoshuaBatty — APPROVED** [review]

**Pull request merged**